### PR TITLE
New cosmetic setting : Invert Y Axis in first person camera

### PR DESCRIPTION
**Pull request by GSKirox**
New cosmetic setting that enables to invert the Y axis when in first person camera.

invert

Obviously this isn't aimed towards anyone already familiar with the game, but the demand has came sometimes from players that are more used to modern camera controls. 

This is achieved by checking the current "mode" of the camera. See https://github.com/zeldaret/oot/blob/master/include/z64camera.h#L207 for all possible states documented.
There was actually already a parameter named camera_mode in the current z64_game structure in rando, which got renamed in camera_setting to be closer to decomp (see https://github.com/zeldaret/oot/blob/master/include/z64camera.h#L137 for what it refers to).

Y inputs will be inverted in the following camera_mode states : 
- First person camera (c-up)
- Aiming with the bow or hookshot/longshot
- Aiming with the slingshot
- Aiming with the boomerang

### CODE REVIEW

**Comment by ETR-BTF** [comment]
Weirdshotting is still possible. I'm not sure if it's different. Someone would have to do frame analysis for that. Hookshot/Bow extension also still works, but you also turn 180 degrees when doing it. A minor inconvenience and time loss, but not game-breaking. I can't think of anything else to test.

Also, doesn't this actually _uninvert_ the Y-axis?

**Comment by GSKirox** [comment]
I guess yeah? I'm too used to it at this point to notice which way is right lol
You can interpret it as "Invert from the normal state of the game" :D

**Comment by r0bd0g** [comment]
This is the type of setting that ought to be able to be set by individual players, but it isn't cosmetic. You might need to come up with some other name for a setting like this. Why only a setting for Y axis? You might want to label both settings "invert" and have Y axis enabled by default.

**Comment by GSKirox** [comment]
I think it's more or less standard to only give the option for Y axis ? 
Inverting X axis is useful in free camera games, but i don't think it's used for first person cameras.

I agree it feels out of place in the Cosmetics tab though.

**Comment by cjohnson57** [comment]
For the purpose of what cosmetic options are it belongs there, similar to default targeting option

**Comment by cjohnson57** [comment]
> Also, doesn't this actually uninvert the Y-axis?

Yeah I'm wondering if there's a way to make that clearer to the user in a succinct way.
Something like "By default, up on the joystick looks down. With this setting, up on the joystick will look up" but I feel like that could be worded better

**Comment by cjohnson57** [comment]
>  Why only a setting for Y axis? You might want to label both settings "invert" and have Y axis enabled by default.

Literally no one has ever asked to have a way to invert the X axis lol

File: ASM/c/invertYaxis.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    if (CFG_INVERT_YAXIS_IN_FIRST_PERSON_CAMERA) {
+        // Never if a choice textbox is on screen.
+        if (z64_MessageGetState(((uint8_t *)(&z64_game)) + 0x20D8) == 4) {
+            return;
+        }
+        if (z64_game.camera_mode == 0x06 ||
+            z64_game.camera_mode == 0x07 ||
+            z64_game.camera_mode == 0x0A ||
+            z64_game.camera_mode == 0x0B) {
```
Comment (cjohnson57) [inline ASM/c/invertYaxis.c]: Might seem a little hypocritical considering how many unexplained "magic numbers" we have like this in the codebase, but I'd appreciate a comment labelling each of the states here so I can see why these are being checked for

Comment (GSKirox) [reply ASM/c/invertYaxis.c]: No you're right, it was lazy on my part.
I added the whole enum from decomp to make the values much clearer.

**Comment by ETR-BTF** [comment]
> > Also, doesn't this actually uninvert the Y-axis?
> 
> Yeah I'm wondering if there's a way to make that clearer to the user in a succinct way. Something like "By default, up on the joystick looks down. With this setting, up on the joystick will look up" but I feel like that could be worded better

"Uninverts the vertical inputs while in first person mode (e.g. C-Up, Slingshot, Hookshot), so that moving the stick up will move the camera up and ditto for down." I think it's important to use the word vertical since technically which axis is the Y-axis is a matter of convention and not universal (even tho Y is typically reserved for vertical).

I do think it should be called "Uninvert Y-Axis" because that's what it does.

**Comment by cjohnson57** [comment]
> I do think it should be called "Uninvert Y-Axis" because that's what it does.

I agree, either that or it should keep its current name and be set to true by default, with more clarification in the tooltip for if someone thinks "Do they mean invert it from normal behavior or invert it in general?"

The current name/description is confusing to new players imo

**Review by cjohnson57 — CHANGES_REQUESTED** [review]
Requesting to change the name/description or keep the name but make true by default (+ inverting value checks obvs)

**Comment by GSKirox** [comment]
Ok for renaming !
Not a big fan of having new options at true by default, it's always a bit confusing imo so should be a last resort, but it's probably a personal preference.

**Review by cjohnson57 — APPROVED** [review]

**Pull request merged**